tests/testthat/test_nullModel.R

```diff
@@ -28,6 +28,28 @@ test_that("design matrix with missing reference level", {
     expect_equal(colnames(nm$model.matrix)[2], "by")
 })
 
+test_that("design matrix with collinear covariates", {
+    set.seed(20); a <- rnorm(10)
+    set.seed(21); c <- sample(1:10, 10, replace=TRUE)
+    dat <- data.frame(a=a,
+                      b=c(rep("a",5), rep("b", 5)),
+                      c=c,
+                      d=sample(1:10, 10, replace=TRUE),
+                      stringsAsFactors = FALSE)
+    dat$e <- dat$c
+    dat$f <- dat$c + 2 * dat$d
+    # No failure with no covariates.
+     expect_silent(nm <- fitNullModel(dat, outcome="a"))
+     # No failure with no colinear covariates.
+     expect_silent(fitNullModel(dat, outcome="a", covars = c("b", "c", "d")))
+     # Simple case where one covariate equals another.
+     expect_error(fitNullModel(dat, outcome="a", covars = c("b", "c", "d", "e")),
+                  "collinear")
+     # Slightly more complicated case involving linear combination of more than one covariate.
+     expect_error(fitNullModel(dat, outcome="a", covars = c("b", "c", "d", "f")),
+                  "collinear")
```

Comment:
I decided to test the behavior of `fitNullmodel`, which is the exported, instead of `createDesignMatrix`, where the actual check is implemented -- following https://r-pkgs.org/tests.html which says "Focus on testing the external interface to your functions". The tests also pass when calling `createDesignMatrix` instead, so I can change it if desired.